src/Symfony/Component/VarDumper/Caster/DOMCaster.php

```diff
@@ -154,8 +155,10 @@ public static function castDocument(\DOMDocument $dom, array $a, Stub $stub, $is
             'preserveWhiteSpace' => $dom->preserveWhiteSpace,
             'recover' => $dom->recover,
             'substituteEntities' => $dom->substituteEntities,
-            "\0~\0xml" => $dom->saveXML(),
         );
+        if (!($filter & Caster::EXCLUDE_VERBOSE)) {
+            $a += array(Caster::PREFIX_VIRTUAL.'xml' => $dom->saveXML());
+        }
 
         $dom->formatOutput = $formatOutput;
```

Comment:
I suggest moving the changing of ``formatOutput`` and its resetting to the previous value inside the ``if (!($filter & Caster::EXCLUDE_VERBOSE)) {`` check as it is needed only for the ``saveXML`` call AFAIK